codes/JavaScript/euclids-algorithm.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Calculates the greatest common divisor of 2 numbers using the Euclid's algorithm
+ * @param {number} a first number
+ * @param {number} b second number
+ * @returns {number} the greatest common divisor
+ */
+function euclid(a, b) {
```

Comment:
What if we use recursion instead like
```js
function euclid(a, b) {
	if(a === 0) return b;
        if(b === 0) return a;
        return euclid(b, a % b);
}
```
Your code runs fine, anyhow

Reply:
I think that both ways are basically the same regarding efficiency, I used this approach because it shows the initial algorithm better imo

Reply:
Maybe display both methods? Just so you know people can reference it better?

Reply:
rhygg's approach is way simpler to understand than yours, i don't get what you mean

Reply:
I meant that the function I wrote is closer to what I would do if I did it by hand

Reply:
I understand, but i'm sure writing both methods wouldn't be a hassle. Commit and i'll push :smile: